src/components/AddBlogForm/AddBlogForm.jsx

```diff
@@ -0,0 +1,117 @@
+import { useState } from "react";
+import { useDispatch } from "react-redux";
+import { Field, Form } from "react-final-form";
+import PropTypes from "prop-types";
+import { v4 as uuidv4 } from "uuid";
+import useAuth from "hooks/useAuth";
+import { postBlog } from "features/blogs/blogsSlice";
+import ImageDnD from "components/ImageDnD/ImageDnD";
+import Button from "components/Button/Button";
+import SelectBox from "components/SelectBox/SelectBox";
+import "./AddBlogForm.scss";
+
+const AddBlogForm = ({ setIsAddBlogFormOpen }) => {
+  const [files, setFiles] = useState([]);
+  const dispatch = useDispatch();
+  const { authUser } = useAuth();
+
+  const onSubmit = async (blog) => {
+    blog.id = uuidv4();
+    blog.authorId = authUser.id;
+    blog.createdAt = new Date().toISOString();
+    const imagePreview = await convertToBase64(files[0]);
```

Comment:
Can this fail? Let's add a try catch here

Reply:
Yes @hhu-wtag bhai, this can fail. I added try catch.